### PR TITLE
test(msteams): Add tests for building group card

### PR DESCRIPTION
**Pull request by 97varun**
Add tests for build_group_card which builds issues cards. Does NOT test all visual aspects of the card. Only ensures that certain important elements are present and the basic structure is correct.

### CODE REVIEW

File: tests/sentry/integrations/msteams/test_message_builder.py [hunk 1]
```diff
@@ -175,3 +213,143 @@ def test_unlink_indentity_message(self):
         assert 1 == len(unlink_identity_card["body"])
         assert 1 == len(unlink_identity_card["actions"])
         assert "test-url" == unlink_identity_card["actions"][0]["url"]
+
+    def test_issue_message_builder(self):
+        self.event1.data["metadata"].update({"value": "some error"})
```
Comment (97varun) [inline tests/sentry/integrations/msteams/test_message_builder.py:218]: To add description to the issue, type has to be "error" and "metadata.value" will be the description.

File: tests/sentry/integrations/msteams/test_message_builder.py [hunk 2]
```diff
@@ -36,6 +45,35 @@ class MSTeamsMessageBuilderTest(TestCase):
...
     """
 
+    def setUp(self):
+        self.user = self.create_user(is_superuser=False)
+        owner = self.create_user()
+        self.org = self.create_organization(owner=owner)
+
+        self.integration = Integration.objects.create(
+            provider="msteams",
+            name="Fellowship of the Ring",
```
Comment (armenzg) [inline tests/sentry/integrations/msteams/test_message_builder.py:55]: Just listening to [this](https://www.youtube.com/watch?v=CahOLfYxiq0)!! 🎧 ❤️ 💍

File: tests/sentry/integrations/msteams/test_message_builder.py [hunk 3]
```diff
@@ -36,6 +45,35 @@ class MSTeamsMessageBuilderTest(TestCase):
...
+            external_id="f3ll0wsh1p",
+            metadata={
+                "service_url": "https://smba.trafficmanager.net/amer",
+                "access_token": "y0u_5h4ll_n07_p455",
+                "expires_at": int(time.time()) + 86400,
+            },
```
Comment (armenzg) [inline tests/sentry/integrations/msteams/test_message_builder.py:61]: What's this metadata for?

File: tests/sentry/integrations/msteams/test_message_builder.py [hunk 4]
```diff
@@ -36,6 +45,35 @@ class MSTeamsMessageBuilderTest(TestCase):
...
+        )
+        OrganizationIntegration.objects.create(organization=self.org, integration=self.integration)
+
+        self.project1 = self.create_project(organization=self.org)
+        self.event1 = self.store_event(
+            data={"message": "oh no"},
+            project_id=self.project1.id,
+        )
+        self.group1 = self.event1.group
```
Comment (armenzg) [inline tests/sentry/integrations/msteams/test_message_builder.py:70]: This is a group of cards; right?

File: tests/sentry/integrations/msteams/test_message_builder.py [hunk 5]
```diff
@@ -36,6 +45,35 @@ class MSTeamsMessageBuilderTest(TestCase):
...
+
+        self.rules = [
```
Comment (armenzg) [inline tests/sentry/integrations/msteams/test_message_builder.py:72]: Are these alert rules?

**Review by armenzg — APPROVED** [review]
Good job @97varun !

File: tests/sentry/integrations/msteams/test_message_builder.py (hunk 4 above)
Comment (97varun) [reply tests/sentry/integrations/msteams/test_message_builder.py:70]: Nope, this is a Sentry `group` which, if I understand correctly, is a group of issues.

File: tests/sentry/integrations/msteams/test_message_builder.py (hunk 3 above)
Comment (97varun) [reply tests/sentry/integrations/msteams/test_message_builder.py:61]: It's not necessary for this test I guess, let me see if I can get rid of it without making the integration object unhappy.

Comment (97varun) [reply tests/sentry/integrations/msteams/test_message_builder.py]: Removed.

**Pull request merged**